### DISCUSSION TITLE
Automatically remove synced defaultValues from code

### DISCUSSION THREAD

**Discussion by noinskit** [tolgee/tolgee-platform#2425] [Show and tell]
I'm converting a website with UI strings inlined with code to Tolgee and I found it very useful to automatically remove the `defaultValue` argument from `t()` invocations (just after they've been synced to Tolgee with tolgee-cli).
I think it would make a nice addition to tolgee-cli.

FWIW this is the code I've been using (using ts-morph), although I'm aware it cannot be used directly due to multiple languages/platforms supported by tolgee-cli:
```
import {Project, SyntaxKind} from "ts-morph";

const project = new Project();
project.addSourceFilesAtPaths("*/src/**/*.{ts,tsx}");

let edits = 0;

project.getSourceFiles().forEach(sourceFile => {
    console.debug(`Processing ${sourceFile.getFilePath()}...`);

    sourceFile.forEachDescendant(node => {
        const callExpression = node.asKind(SyntaxKind.CallExpression);
        if (!callExpression) return;

        const expression = callExpression.getExpression();
        if (expression.getText() === "t" && callExpression.getArguments().length >= 2) {
            const secondArg = callExpression.getArguments()[1];
            if (secondArg.getKind() === SyntaxKind.StringLiteral) {
                edits++;
                callExpression.removeArgument(1);
            }
        }
    });
});

project.save().then(() => {
    console.log(`Transformation complete (${edits} edits)`);
});
```

**Comment by cyyynthia** [comment]
I'm wondering, what's the use-case for dropping the default string for language keys? Is it to avoid the "unnecessary" declaration of the default value or does it have other benefits?

**Reply by noinskit** [reply]
It's not about performance or some hidden benefits, it's just my idea for resilience against bugs.

I want to have one source of truth for the main language and then the only reasonable place is in the Tolgee platform.
Otherwise it's duplication and it's not even that easy to keep them in sync. As far as I can tell, tolgee-cli does not warn about code defaults being different than their counterparts in the platform (I tried `sync` and `compare` subcommands).

And I don't think these defaults give me much as last-resort fallbacks, because they're likely to be in the wrong language (one hardcoded language, not what user wants) and might cause more harm by hiding the problem.


On the other hand, I'm finding these default values great for converting a non-Tolgee website to Tolgee.
My workflow is (file-by-file):

1) Convert raw strings to Tolgee strings, e.g. `foo` -> `t("dir1.dir2.SomeClass.header", "foo")`. This can be greatly assisted by AI tools: I'm using Sonnet 3.5 via Cody IDE plugin and it's smart enough to come up with good keys and even use placeholders/properties correctly.

2) Run `tolgee sync` to upload these defaults to the Tolgee platform.

3) Nuke the defaults using the above script.
